### PR TITLE
Spike/32341 spike azure functions filesystem access

### PR DESCRIPTION
**Pull request by jon-shipley**
Spike to investigate filesystem access on Azure Functions

- [x] test that we can create largish files (525 MB)
- [x] test that we can create a zip archive
- [x] test file deletion
- [x] test ability to report disk space on the OS's tmpdir

### CODE REVIEW

File: _spikes-poc/function-fs-access/functions/long-runner/v1.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+let logger
+
+function delay (ms) {
+  return new Promise(function (resolve) {
+    setTimeout(resolve, ms)
+  })
+}
+
+const v1 = {
+  process: async function (loggerArg) {
```
Comment (GeorgeChatzigiannis) [inline _spikes-poc/function-fs-access/functions/long-runner/v1.js]: @jon-shipley can you provide some details about the purpose of this function?
what is the benefit here? 🤔 
Are we just verifying that a function execution time can reach the 2hrs milestone?

File: _spikes-poc/function-fs-access/functions/test-fs-access/v1.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    stream.on('error', function (error) {
+      console.error(`Error reading file ${fileName}: ${error.message}`)
+      reject(error)
+    })
+
+    stream.on('end', function () {
+      logger(`End event: Contents of ${fileName}: ${lines.join('\n')}`)
+      resolve()
+      data = null
```
Comment (GeorgeChatzigiannis) [inline _spikes-poc/function-fs-access/functions/test-fs-access/v1.js]: @jon-shipley is setting data to null necessary?

Comment (jon-shipley) [reply _spikes-poc/function-fs-access/functions/test-fs-access/v1.js]: Probably not, but it forces memory to be freed.

File: _spikes-poc/function-fs-access/functions/long-runner/v1.js (hunk 1 above)
Comment (jon-shipley) [reply _spikes-poc/function-fs-access/functions/long-runner/v1.js]: Yes, exactly that

On the Consumption plan functions time out after 10 minutes at the maximum.  On the App service plan functions can run for as long as you want.  This was just proving that.

**Pull request merged**